contrib/parseq-examples/src/main/java/com/linkedin/parseq/example/batching/BatchingClientExample.java

```diff
@@ -0,0 +1,70 @@
+/* $Id$ */
+package com.linkedin.parseq.example.batching;
+
+import java.util.List;
+
+import com.linkedin.parseq.Engine;
+import com.linkedin.parseq.Task;
+import com.linkedin.parseq.batching.BatchEntry;
+import com.linkedin.parseq.batching.BatchingStrategy;
+import com.linkedin.parseq.batching.BatchingSupport;
+import com.linkedin.parseq.example.common.AbstractExample;
+import com.linkedin.parseq.example.common.ExampleUtil;
+import com.linkedin.parseq.promise.Promises;
+
+
+/**
+ * @author Jaroslaw Odzga (jodzga@linkedin.com)
+ */
+public class BatchingClientExample extends AbstractExample {
+
+  public static class ExampleBatchingStrategy extends BatchingStrategy<Long, Long> {
+
+    @Override
+    public Long groupForKey(Long key) {
```

Comment:
I agree. Perhaps we could provide BatchingStrategy that would be most general and would not do any grouping and a subclass GroupingBatchingStrategy that would be more convenient for grouping (it would contain changes you are mentioning).